C/STrace/EtwLogger.cpp

```diff
@@ -0,0 +1,460 @@
+#include "EtwLogger.h"
+
+#include "vector.h"
+
+#define va_copy(dest, src) (dest = src)
+
+// Cache of all created providers.
+MyVector<detail::EtwProvider> g_ProviderCache;
+
+typedef enum _ETW_FIELD_TYPE
+{
+	EtwFieldNull,
+	EtwFieldUnicodeString,
+	EtwFieldAnsiString,
+	EtwFieldInt8,
+	EtwFieldUInt8,
+	EtwFieldInt16,
+	EtwFieldUInt16,
+	EtwFieldInt32,
+	EtwFieldUInt32,
+	EtwFieldInt64,
+	EtwFieldUInt64,
+	EtwFieldFloat,
+	EtwFieldDouble,
+	EtwFieldBool32,
+	EtwFieldBinary,
+	EtwFieldGuid,
+	EtwFieldPointer,
+	EtwFieldFiletime,
+	EtwFieldSystemTime,
+	EtwFieldSid,
+	EtwFieldHexInt32,
+	EtwFieldHexInt64,
+	EtwFieldPid = (EtwFieldInt32 | 0x05 << 8),
+} ETW_FIELD_TYPE;
+
+namespace detail
+{
+
+#pragma region EtwProvider
+
+EtwProvider::EtwProvider(LPCGUID providerGuid) : m_guid{ providerGuid }, m_regHandle{}, m_providerMetadataDesc{}, m_events{}
+{
+}
+
+EtwProvider::EtwProvider(EtwProvider&& other)
+{
+	m_guid = move(other.m_guid);
```

Comment:
std::move only allows selection of a move constructor, for built in types this is not necessary as they don't have move semantics. You can remove std::move from everything here that isn't a class you've written yourself that supports move semantics.